### PR TITLE
Implements block-level node finding, and node splitting.

### PR DESCRIPTION
**Pull request by diegoreymendez**
This PR implements several logical instruments we'll need to update the DOM.

- Block-level node finding basically means we can find all lowest (hierarchy-wise) block-level nodes intercepting a range.
- Splitting nodes and their children using a range.

**How to test:**
- Since this PR is mostly non-hooked logical methods, there's not much that can be tested from the GUI yet.
- Run the unit tests and make sure they succeed.

### CODE REVIEW

File: Classes/Private/Libxml2/Data/ElementNode.swift [hunk 1]
```diff
@@ -215,6 +366,322 @@ extension Libxml2 {
...
+            }
+
+            children.removeAtIndex(childIndex)
+            children.insertContentsOf(newChildren, at: childIndex)
+        }
+
+        /// Removes the specified child nodes.  Only updates their parents if specified.
+        ///
+        /// - Parameters:
+        ///     - chldren: the child nodes to remove.
```
Comment (SergioEstevao) [inline Classes/Private/Libxml2/Data/ElementNode.swift]: small type in parameter name.

File: Classes/Private/Libxml2/Data/ElementNode.swift [hunk 2]
```diff
@@ -215,6 +366,322 @@ extension Libxml2 {
...
+                    break
+                } else {
+                    break
+                }
+            }
+            
+            return result
+        }
+
+        /// Retrieves all child nodes positioned before a specified location.
```
Comment (SergioEstevao) [inline Classes/Private/Libxml2/Data/ElementNode.swift]: all children nodes?

**Comment by SergioEstevao** [comment]
Looking good @diegoreymendez . Just made some small notes about typos in the comments. After you fix those :shipit:

File: Classes/Private/Libxml2/Data/ElementNode.swift (hunk 2 above)
Comment (diegoreymendez) [inline Classes/Private/Libxml2/Data/ElementNode.swift]: So in theory it would be either "all children" (without the word "nodes") or "all child nodes".

**Comment by diegoreymendez** [comment]
Thanks @SergioEstevao!

**Pull request merged**